### PR TITLE
Allow buffering move, species, ability and item names in tests

### PR DESCRIPTION
**Pull request by Bassoonian**
Move, species, ability and item names used to be hardcoded in test messages, resulting in issues when toggling expanded move names (or when anyone would change any move name for whatever reason). This PR adds the possibility to buffer move, ability, species and item names directly into test messages.

I more than likely missed occurrences of any of the four categories above (I did use a regex to convert everything), so feel free to point those out.

## **Discord contact info**
bassoonian

### CODE REVIEW

File: test/battle/ability/anger_shell.c [hunk 1]
```diff
@@ -49,15 +49,15 @@ SINGLE_BATTLE_TEST("Anger Shell lowers Def/Sp.Def by 1 and raises Atk/Sp.Atk/Spd
         ANIMATION(ANIM_TYPE_MOVE, MOVE_TACKLE, opponent);
         ABILITY_POPUP(player, ABILITY_ANGER_SHELL);
         ANIMATION(ANIM_TYPE_GENERAL, B_ANIM_STATS_CHANGE, player);
-        MESSAGE("Wobbuffet's Defense fell!");
+        MESSAGE(SPECIES_NAME(SPECIES_WOBBUFFET) "'s Defense fell!");
```
Comment (DizzyEggg) [inline test/battle/ability/anger_shell.c:52]: Honestly I would keep `Wobbuffet's`. Writing `SPECIES_NAME(SPECIES_WOBBUFFET) "` every time seems frustrating to do :D

**Comment by mrgriffin** [comment]
Oh! merrp's PR to do automatic decapitalization might make this redundant if/when it's finished? :crossed_fingers:

File: test/battle/ability/anger_shell.c (hunk 1 above)
Comment (AlexOn1ine) [reply test/battle/ability/anger_shell.c:52]: But can you not check multiply mons inside one message this way?

Comment (Bassoonian) [reply test/battle/ability/anger_shell.c:52]: Nope, which is why it got drafted. We bumped into an issue where preproc didn't take variables into account :(

**Comment by Bassoonian** [comment]
> Oh! merrp's PR to do automatic decapitalization might make this redundant if/when it's finished? 🤞

The main issue is extended move names (and, for some users, translations), which the automatic decapitalization doesn't deal with

**Comment by mrgriffin** [comment]
> > Oh! merrp's PR to do automatic decapitalization might make this redundant if/when it's finished? 🤞
> 
> The main issue is extended move names (and, for some users, translations), which the automatic decapitalization doesn't deal with

Ahhh, yeah!

**Comment by Bassoonian** [comment]
Closing this for now until we can think of a better way to cope with this.

**Pull request closed without merging**